src/main/java/cz/sparko/boxitory/service/FilesystemBoxRepository.java

```diff
@@ -75,16 +73,41 @@ private boolean validateFilename(File boxFile) {
         return true;
     }
 
-    private BoxVersion parseBoxFile(File file) {
+    private String getBoxVersionFromFileName(File file) {
         String filename = file.getName();
         List<String> parsedFilename = Arrays.asList(filename.split("_"));
+        return parsedFilename.get(1);
+    }
+
+    private List<BoxVersion> createBoxVersionsFromGroupedFiles(Map<String, List<File>> groupedFiles) {
+        List<BoxVersion> boxVersions = new ArrayList<>();
+        groupedFiles
+                .entrySet()
+                .forEach(
+                    (entry) -> boxVersions.add(createBoxVersion(entry.getKey(), entry.getValue()))
+                );
+        return boxVersions;
+    }
 
-        String name = parsedFilename.get(0);
-        String version = parsedFilename.get(1);
+    private BoxVersion createBoxVersion(String version, List<File> fileList) {
+        List<BoxProvider> boxProviders = new ArrayList<>();
+
+        if (!fileList.isEmpty()) {
+            fileList.forEach(
```

Comment:
this method could be rewritten by single stream like this
`return fileList.stream().map(this::createBoxProviderFromFile).collector(tolist)`
(didnt tested, but should be same behavior).